pkg/common/json_test.go

```diff
@@ -58,6 +58,70 @@ func TestJSONValueResolveFor(t *testing.T) {
 	var resolvedRoles []string
 	_ = json.Unmarshal(resolvedValueAsJSON, &resolvedRoles)
 	assert.DeepEqual(t, resolvedRoles, []string{"user", "admin"})
+
+	// Message with variable placeholder
+	value = JSONValue{Pattern: "Hello, {auth.identity.username}!"}
+	assert.Equal(t, value.ResolveFor(jsonData), "Hello, john!")
+}
+
+func TestReplaceJSONPlaceholders(t *testing.T) {
```

Comment:
Maybe add some tests for "invalid" placeholder specifications? 

Maybe check the CR for invalid fields and reject the creation/update?